lib/js_of_ocaml/typed_array.ml

```diff
@@ -49,47 +51,61 @@ class type ['a, 'b] typedArray = object
 
   method set_fromArray : 'a js_array t -> int -> unit meth
 
-  method set_fromTypedArray : ('a, 'b) typedArray t -> int -> unit meth
+  method set_fromTypedArray : ('a, 'b, 'c) typedArray t -> int -> unit meth
 
-  method subarray : int -> int -> ('a, 'b) typedArray t meth
+  method subarray : int -> int -> ('a, 'b, 'c) typedArray t meth
 
-  method subarray_toEnd : int -> ('a, 'b) typedArray t meth
+  method subarray_toEnd : int -> ('a, 'b, 'c) typedArray t meth
 
-  method slice : int -> int -> ('a, 'b) typedArray t meth
+  method slice : int -> int -> ('a, 'b, 'c) typedArray t meth
 
-  method slice_toEnd : int -> ('a, 'b) typedArray t meth
+  method slice_toEnd : int -> ('a, 'b, 'c) typedArray t meth
 
   (* This fake method is needed for typing purposes.
      Without it, ['b] would not be constrained. *)
-  method _content_type_ : 'b optdef readonly_prop
+  method _content_type_ : ('b * 'c) optdef readonly_prop
 end
 
-type int8Array = (int, Bigarray.int8_signed_elt) typedArray
+type int8Array = (int, int, Bigarray.int8_signed_elt) typedArray
+
+type uint8Array = (int, int, Bigarray.int8_unsigned_elt) typedArray
 
-type uint8Array = (int, Bigarray.int8_unsigned_elt) typedArray
+type int16Array = (int, int, Bigarray.int16_signed_elt) typedArray
 
-type int16Array = (int, Bigarray.int16_signed_elt) typedArray
+type uint16Array = (int, int, Bigarray.int16_unsigned_elt) typedArray
 
-type uint16Array = (int, Bigarray.int16_unsigned_elt) typedArray
+type int32Array = (int32, Int32.t, Bigarray.int32_elt) typedArray
 
-type int32Array = (int32, Bigarray.int32_elt) typedArray
+type uint32Array = (uint32, Int32.t, Bigarray.int32_elt) typedArray
 
-type uint32Array = (int32, Bigarray.int32_elt) typedArray
+type float32Array = (number_t, float, Bigarray.float32_elt) typedArray
 
-type float32Array = (float, Bigarray.float32_elt) typedArray
+type float64Array = (number_t, float, Bigarray.float64_elt) typedArray
 
-type float64Array = (float, Bigarray.float64_elt) typedArray
+type ('bigarray, 'typed_array, 'elt) type' =
```

Comment:
Maybe this could be named kind ? similar to `Bigarray.kind`

Reply:
I also like `kind` better (I updated the code), I’m just wondering whether it can be annoying in practice that both the `Bigarray` and `Typed_array` modules contain a type `kind`.